bigcommerce/widget-builder
# MERC-5808 Support graphql

@bigcommerce/merc-team 

## Technical summary
- Running `widget-builder` with `--gen-query-params` will create `queryParams.json` as long as `queryParamsBuilder.json` has been provided in the directory beforehand.
- Modifying `config.json` will regenerate `queryParams.json` and rerender the widget accordingly.

## Steps to test
- checkout this PR and `npm run install-cli` in `widget-builder`
- ` cd ../content/app/templates/sd-product-list/`
- create `queryParamsBuilder.json` with the following content:
```
{
  "productIds": {
    "reads": "products.*.productId",
    "type": "Int!"
  }
}
```
- run this command in `widget-builder`: 
```
WIDGET_BUILDER_AUTH_ID=yourprodstoreauthid WIDGET_BUILDER_AUTH_TOKEN=yourprodstoreauthtoken WIDGET_BUILDER_STORE_HASH=yourprodstorehash widget-builder --gen-query-params --gen-config
```
- change productids in `config.json`
- see the awesomeness!
![Screen Shot 2020-01-13 at 2 10 04 PM](https://user-images.githubusercontent.com/45639285/72296445-69c23a00-360e-11ea-8cf0-ffc9882e58eb.png)
